### ISSUE TITLE
OpenAPI: Parsing of JSON downloaded from a URL fails if no schemes element is present

### ISSUE THREAD

**Issue by robertofpaf** [zaproxy/zaproxy#3433]
The current version of OpenAPI add-on supports only yaml format as it uses version 1.0.19 of swagger-parser. Looking at the release notes for that project in version 1.0.23 support for JSON format has been added, so it would be nice if that one (or the latest 1.0.26) could be taken into use as springfox (an add-on for spring-boot that generates OpenAPI documentation) doesn't support the yaml format at the moment.

**Comment by psiinon** [comment]
I'm confused.
The OpenAPI add-on seems to work fine with JSON - we have regression tests for both: https://github.com/zaproxy/zap-extensions/tree/alpha/test/resources/org/zaproxy/zap/extension/openapi and https://github.com/zaproxy/zap-extensions/blob/alpha/test/org/zaproxy/zap/extension/openapi/OpenApiUnitTest.java and I've tested it manually with JSON as well.
Unless I'm missing something ;)
I'm all for keeping libs up to date of course, but I'm not sure that just updating the lib will add new functionality in this case.

**Comment by robertofpaf** [comment]
Ah my bad, I misread the release notes for swagger-parser 1.0.23, they didn't add support for JSON but for JSON References. Actually the problem I'm facing is more subtle: springfox doesn't generate a "schemes" element in the JSON file, and that's according to specs (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md):
schemes 	[string] 	The transfer protocol of the API. Values MUST be from the list: "http", "https", "ws", "wss". If the schemes is not included, the default scheme to be used is the one used to access the Swagger definition itself.

I tried to import the OpenAPI definition from a URL but it fails with a NPE:
```
15917 [ZAP-Import-OpenAPI-1] WARN org.zaproxy.zap.extension.openapi.ExtensionOpenApi  - 
java.lang.NullPointerException
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.readOpenAPISpec(SwaggerConverter.java:71)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getRequestModels(SwaggerConverter.java:54)
	at org.zaproxy.zap.extension.openapi.ExtensionOpenApi$3.run(ExtensionOpenApi.java:192)
```
the referred row in SwaggerConverter is `for (Scheme scheme : swagger.getSchemes()) {` so the converter assumes that the schemes element is always present, not following the specs.

**Comment by psiinon** [comment]
Np :) Yeah, should definitely fix this asap.
Thanks for reporting it.

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#851.

**Comment by thc202** [comment]
Released in version 4 of OpenAPI Support add-on.
